site/en/blog/prerender-pages/index.md

```diff
@@ -0,0 +1,263 @@
+
+### Using the Speculation rules API to prerender pages
+
+For the third prerender option, web developers can insert JSON instructions onto their pages to inform the browser about which URLs would benefit from prerendering:
+
+```html
+<script type="speculationrules">
+{
+  "prerender": [
+    {
+      "source": "list",
+      "urls": ["next.html", "next2.html"]
+    }
+  ]
+}
+</script>
+```
+
+The Speculation rules API is planned to be expanded beyond this simple example with the addition of prefetch, [scores](https://github.com/WICG/nav-speculation/blob/main/triggers.md#scores) (for example, the likelihood of a navigation), and syntax to implement [document rules](https://github.com/WICG/nav-speculation/blob/main/triggers.md#document-rules) (for example, matching `href` patterns). For now, only the above syntax is supported in Chrome.
+
+For the initial launch in Chrome 108, prerender is restricted to same-origin pages, opened within the same tab. In Chrome 109 we expect to expand this to allow [prerendering same-site cross-origin pages](https://chromestatus.com/feature/4899735257743360) (for example, `https://a.example.com` could prerender a page on `https://b.example.com`, where the other site has opted in to same-site, cross-origin prerendering). We are also exploring options to [enable prerendering in new tabs](https://bugs.chromium.org/p/chromium/issues/detail?id=1350676).
+
+You can feature detect Speculation rules support with standard HTML checks:
+
+```js
+if (HTMLScriptElement.supports && HTMLScriptElement.supports('speculationrules')) {
+  console.log('Your browser supports the Speculation rules API.');
+}
+```
+
+Speculation rules can be:
+
+* Statically inserted in the page's HTML. For example a news media site, or a blog may prerender the newest article, if that is often the next navigation for a large proportion of users.
+* Dynamically inserted into the page by JavaScript. This could be based on application logic, personalized to the user, or on certain user actions such as hovering over, or clicking down on a link—as many libraries have done in the past with `preconnect`, `prefetch`, or even `preload`.
+
+Speculation rules can be added anywhere in the main page and do not need to be in the `<head>` element. Speculation rules in subframes are not acted upon, and speculation rules in prerendered pages, are only acted upon once the page is activated.
+
+Below is a simple example of adding a `prerender` speculation rule:
+
+```js
+if (HTMLScriptElement.supports &&
+    HTMLScriptElement.supports('speculationrules')) {
+    const specScript = document.createElement("script");
+    specScript.type = 'speculationrules';
+    specRules = {
+      'prerender': [
+        {
+          source: 'list',
+          urls: ['/next.html'],
+        },
+      ],
+    };
+    specScript.textContent = JSON.stringify(specRules);
+    console.log('added speculation rules to: next.html');
+    document.body.append(specScript);
+}
+```
```

Comment:
```suggestion
```js
if (HTMLScriptElement.supports && HTMLScriptElement.supports('speculationrules')) {
  const specScript = document.createElement('script');
  specScript.type = 'speculationrules';
  specRules = {
    'prerender': [
      {
        source: 'list',
        urls: ['/next.html'],
      },
    ],
  };
  specScript.textContent = JSON.stringify(specRules);
  console.log('added speculation rules to: next.html');
  document.body.append(specScript);
}
```
```